boxoffice/views/admin_order.py

```diff
@@ -80,7 +80,48 @@ def admin_orders(item_collection):
 @load_models(
     (Order, {'id': 'order_id'}, 'order'),
     permission='org_admin'
-    )
+)
 def admin_order(order):
     line_items = LineItem.query.filter(LineItem.order == order, LineItem.status.in_([LINE_ITEM_STATUS.CONFIRMED, LINE_ITEM_STATUS.CANCELLED])).all()
     return jsonify(line_items=format_line_items(line_items))
+
+
+def jsonify_order(order_dict):
+    org = {
+        "title": order_dict['org'].title,
+        "name": order_dict['org'].name
+    }
+    order = {
+        "id": order_dict['order'].id,
+        "buyer_fullname": order_dict['order'].buyer_fullname,
+        "buyer_email": order_dict['order'].buyer_email,
+        "buyer_phone": order_dict['order'].buyer_phone,
+        "invoice_no": order_dict['order'].invoice_no,
+        "receipt_url": url_for('receipt', access_token=order_dict['order'].access_token),
+        "assignee_url": url_for('line_items', access_token=order_dict['order'].access_token)
+    }
+    ic = {
+        'id': order_dict['order'].item_collection_id
+    }
+    return jsonify(
+        org=org,
+        ic=ic,
+        order=order,
+        line_items=format_line_items(order_dict['order'].line_items)
+    )
+
+
+@app.route('/admin/o/<org_name>/order/<receipt_no>', methods=['GET', 'POST'])
+@lastuser.requires_permission(('siteadmin', 'org_admin'))
+@lastuser.requires_login
+@render_with({'text/html': 'index.html.jinja2', 'application/json': jsonify_order})
+def admin_org_order(org_name, receipt_no):
+    org = Organization.query.filter_by(name=org_name).one_or_404()
+    try:
+        int(receipt_no)
+    except:
+        abort(400)
```

Comment:
This can be removed if you move it to the URL, but you may have to check what happens if the parameter is not an integer. My guess, you'll get a 404 instead of a 400.